src/Ushahidi/Modules/V5/Repository/Permissions/PermissionsRepository.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+namespace Ushahidi\Modules\V5\Repository\Permissions;
+
+use Ushahidi\Modules\V5\Models\Permissions;
+use Illuminate\Pagination\LengthAwarePaginator;
+
+interface PermissionsRepository
+{
+ /**
+     * This method will fetch all the Permissions for the logged user from the database utilising
+     * Laravel Eloquent ORM and return them as an array
+     * @param int $limit
+     * @param int $skip
+     * @param string $sortBy
+     * @param string $order
+     * @return Permissions[]
+     */
+    public function fetch(
+        int $limit,
+        int $skip,
+        string $sortBy,
+        string $order,
+        array $search_data
+    ): LengthAwarePaginator;
```

Comment:
We should probably return Collection, and Controller should convert that to the LengthAwarePaginator.